### DISCUSSION TITLE
Combining same channels giving file name collision error

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#4002] [Q&A]
Hi,

I have a channel of `tuple (val, file)`. I assigned it into two channels `ch1, ch2`. So, both channels have same element and then combined them using `combine` operator. Because, I need the same element twice. But, when I use the combined channel in a process I am getting error that: `input file name collision -- There are multiple files for each of the following file names.` Any help?

Also if I use separately `ch1` and `ch2` like: 
```
input:
tuple val(st1), path(mng1) from ch1
tuple val(st2), path(mng2) from ch2
```
The same message comes out.

**Comment by bentsherman** [comment]
Hi @zillurbmb51 , when you provide a path input to a process, Nextflow stages the path into the task work directory. So if you provide multiple paths with the same filename, you will get a name conflict. When you combine these two channels, you should remove any combinations that have the same file twice. Or, you should stage them under different names, for example:

```groovy
input:
tuple val(st), path('mng1.txt'), path('mng2.txt') // or whatever the file extension is
```
```
